Python/bltinmodule.c

```diff
@@ -2419,7 +2419,7 @@ builtin_sum_impl(PyObject *module, PyObject *iterable, PyObject *start)
                 Py_DECREF(item);
                 continue;
             }
-            if (PyLong_CheckExact(item)) {
+            if (PyLong_Check(item)) {
```

Comment:
Why is this not `PyLong_CheckExact(item) || PyBool_Check(item)` like the one above?

Reply:
Because why not? This extends the optimization to other int subclasses.

There is a reason why `PyLong_Check(item)` is not used above: because `__radd__` method in the int subclass takes precedence. But for float+int it is not used.